corehq/apps/data_interfaces/templates/data_interfaces/partials/auto_update_rule_list.html

```diff
@@ -26,9 +26,16 @@
         <tbody data-bind="foreach: paginatedRules">
           <tr>
             <td>
-              <a data-bind="attr: {href: edit_url}">
-                <strong data-bind="text: name"></strong>
-              </a>
+              <span data-bind="if: upstream_id">
+                <a data-bind="attr: {href: view_url}">
+                  <strong data-bind="text: name"></strong>
+                </a>
+              </span>
+              <span data-bind="ifnot: upstream_id">
+                <a data-bind="attr: {href: edit_url}">
+                  <strong data-bind="text: name"></strong>
+                </a>
+              </span>
```

Comment:
The code has evolved now where it's not just switching based on `upstream_id`, but rather `upstream_id` and the lock state. Because of that, I don't think it's a good idea to combine expression logic with knockout conditionals, but I do think its reasonable to create a hybrid variable to reduce the duplication. Looking over the javascript, it looks like the rules are read directly from the server, with no view model set up to handle the individual rules. I'll play with it to see if its reasonable to create a rule view model that has a computed property for the correct URL.

Reply:
Latest commit has a new view model for individual update rules which eliminates this duplication